pkg/forwarder/internal/retry/on_disk_retry_queue.go

```diff
@@ -95,13 +95,15 @@ func (s *onDiskRetryQueue) Store(transactions []transaction.Transaction) error {
 		_ = os.Remove(file.Name())
 		return err
 	}
-	defer file.Close()
-
 	s.currentSizeInBytes += bufferSize
 	s.filenames = append(s.filenames, file.Name())
 	s.telemetry.setFileSize(bufferSize)
 	s.telemetry.setCurrentSizeInBytes(s.GetDiskSpaceUsed())
 	s.telemetry.setFilesCount(s.getFilesCount())
+	err = file.Close()
+	if err != nil {
+		return fmt.Errorf("could not close file [%s]: %w", file.Name(), err)
+	}
```

Comment:
Also with the current version you edit `s` then possibly return an error, which I'm not sure is expected (eg. it might be reasonable to assume that if an error is returned, no file was appended to the list).

Reply:
I prefer to avoid using a named return value as this change could affect the signature of the function.  It could impact other parts of the code and bring new errors.

> Also with the current version you edit s then possibly return an error, which I'm not sure is expected (eg. it might be reasonable to assume that if an error is returned, no file was appended to the list).

That's a good catch. I tried to mimic the current behavior of the function by putting the `Close()` at the end but I agree that editing `s` before calling `Close()` might be inconsistent, especially with `s.filenames = append(s.filenames, file.Name())`.

Thanks for the insights !